packages/react/block.ts

```diff
@@ -9,71 +9,71 @@ import { MapSet$, MapHas$ } from '../million/constants';
 import { queueMicrotask$ } from '../million/dom';
 import type { Options, MillionProps, MillionPortal } from '../types';
 import { processProps, unwrap } from './utils';
-import { Effect, RENDER_SCOPE, REGISTRY, SVG_RENDER_SCOPE } from './constants';
+import { useContainer, useNearestParent } from './its-fine';
+import { Effect, REGISTRY } from './constants';
 
 export const block = <P extends MillionProps>(
   fn: ComponentType<P> | null,
-  options: Options | null | undefined = {}
+  options: Options | null | undefined = {},
 ) => {
   let block: ReturnType<typeof createBlock> | null = options?.block;
-  const defaultType = options?.svg ? SVG_RENDER_SCOPE : RENDER_SCOPE;
-
   if (fn) {
     block = createBlock(
       fn as any,
       unwrap as any,
       options?.shouldUpdate,
-      options?.svg
+      options?.svg,
     );
   }
 
   const MillionBlock = <P extends MillionProps>(
     props: P,
-    forwardedRef: Ref<any>
+    forwardedRef: Ref<any>,
   ) => {
+    const container = useContainer<HTMLElement>(); // usable when there's no parent other than the root element
+    const parentRef = useNearestParent<HTMLElement>();
+
     const hmrTimestamp = props._hmr;
-    const ref = useRef<HTMLElement>(null);
     const patch = useRef<((props: P) => void) | null>(null);
     const portalRef = useRef<MillionPortal[]>([]);
 
     props = processProps(props, forwardedRef, portalRef.current);
     patch.current?.(props);
 
     const effect = useCallback(() => {
-      if (!ref.current) return;
+      const target = parentRef.current ?? container.current;
+      if (!target) return;
       const currentBlock = block!(props, props.key);
-      if (hmrTimestamp && ref.current.textContent) {
-        ref.current.textContent = '';
-      }
+      // if (hmrTimestamp && target.textContent) {
```

Comment:
I removed this line because it's not a valid HMR for the new design, I wonder if anyone has any idea around it.

Reply:
maybe we should call patch again?